### PR TITLE
improve the compatibility of "Trainer"

### PR DESCRIPTION
**Pull request by hendrydong**
Description：简要描述这次PR的内容
Delete DEFAULT_CHECK_BATCH_SIZE and make it same with the input batch size.

Main reason: 做出这次修改的原因
It is unnecessary to use DEFAULT_CHECK_BATCH_SIZE, which may cause some conficts with the initialized model.

Checklist  检查下面各项是否完成

Please feel free to remove inapplicable items for your PR.

-	[x] The PR title starts with [$CATEGORY] (例如[bugfix]修复bug，[new]添加新功能，[test]修改测试，[rm]删除旧代码)
-	[x] Changes are complete (i.e. I finished coding on this PR)  修改完成才提PR
-	[x] All changes have test coverage  修改的部分顺利通过测试。对于fastnlp/fastnlp/*的修改，测试代码**必须**提供在fastnlp/test/*。
-	[x] Code is well-documented  注释写好，API文档会从注释中抽取
-	[x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change  修改导致例子或tutorial有变化，请找核心开发人员

Changes: 逐项描述修改的内容
- 去掉 DEFAULT_CHECK_BATCH_SIZE，并将其修改为预先设置的batch_size

Mention: 找人review你的PR

@修改过这个文件的人
@核心开发人员

### CODE REVIEW

**Comment by FengZiYjun** [comment]
@hendrydong 
>It is unnecessary to use DEFAULT_CHECK_BATCH_SIZE, which may cause some conficts with the initialized model.

What conflicts DEFAULT_CHECK_BATCH_SIZE will cause?
Code check mechanism computes the gradients but does NOT change the weights of the model.
It also zeros the gradients before exit.

**Comment by hendrydong** [comment]
> @hendrydong
> 
> > It is unnecessary to use DEFAULT_CHECK_BATCH_SIZE, which may cause some conficts with the initialized model.
> 
> What conflicts DEFAULT_CHECK_BATCH_SIZE will cause?
> Code check mechanism computes the gradients but does NOT change the weights of the model.
> It also zeros the gradients before exit.

Since we provide a batch_size at the beginning of the training time, in some cases, we might use such information to do some reshape or mask, which would be deterministic and well-defined, so that it can clear up some ambiguities. However, DEFAULT_CHECK_BATCH_SIZE make such the design incompatible. Besides, we cannot notice DEFAULT_CHECK_BATCH_SIZE but check the source codes and replacing it with the batch_size would be natural, so removing DEFAULT_CHECK_BATCH_SIZE might be more coding-friendly.

**Comment by FengZiYjun** [comment]
> > @hendrydong
> > > It is unnecessary to use DEFAULT_CHECK_BATCH_SIZE, which may cause some conficts with the initialized model.
> > 
> > 
> > What conflicts DEFAULT_CHECK_BATCH_SIZE will cause?
> > Code check mechanism computes the gradients but does NOT change the weights of the model.
> > It also zeros the gradients before exit.
> 
> Since we provide a batch_size at the beginning of the training time, in some cases, we might use such information to do some reshape or mask, which would be deterministic and well-defined, so that it can clear up some ambiguities. However, DEFAULT_CHECK_BATCH_SIZE make such the design incompatible. Besides, we cannot notice DEFAULT_CHECK_BATCH_SIZE but check the source codes and replacing it with the batch_size would be natural, so removing DEFAULT_CHECK_BATCH_SIZE might be more coding-friendly.

It is good practice that `batch_size` shall never be hard coded in your program, because when it is used it can be always dynamically determined or inferred from other dimensions.  For example, 
`batch_size = x.size(0)` and `x = x.view(-1, seq_len, word_len)`

**Comment by hendrydong** [comment]
> > > @hendrydong
> > > > It is unnecessary to use DEFAULT_CHECK_BATCH_SIZE, which may cause some conficts with the initialized model.
> > > 
> > > 
> > > What conflicts DEFAULT_CHECK_BATCH_SIZE will cause?
> > > Code check mechanism computes the gradients but does NOT change the weights of the model.
> > > It also zeros the gradients before exit.
> > 
> > 
> > Since we provide a batch_size at the beginning of the training time, in some cases, we might use such information to do some reshape or mask, which would be deterministic and well-defined, so that it can clear up some ambiguities. However, DEFAULT_CHECK_BATCH_SIZE make such the design incompatible. Besides, we cannot notice DEFAULT_CHECK_BATCH_SIZE but check the source codes and replacing it with the batch_size would be natural, so removing DEFAULT_CHECK_BATCH_SIZE might be more coding-friendly.
> 
> It is good practice that `batch_size` shall never be hard coded in your program, because when it is used it can be always dynamically determined or inferred from other dimensions. For example,
> `batch_size = x.size(0)` and `x = x.view(-1, seq_len, word_len)`

What I mean is that the DEFAULT_CHECK_BATCH_SIZE is not mentioned before checking the code. Indeed, setting the x.size(0) is practical, users’ unawareness of the  DEFAULT_CHECK_BATCH_SIZE might cause some troubles, including bugs with fixed batch_size.  Besides, the latest PyTorch supports static computation graph, obviously the fixed design will bring less uncertainty and be more stable.
